src/providers/OrgApps.js

```diff
@@ -0,0 +1,50 @@
+import React, { useContext, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import { useApps } from '@aragon/connect-react'
+
+const OrgAppsContext = React.createContext({
+  apps: null,
+  agreementApp: null,
+  disputableVotingApp: null,
+  appsLoading: true,
+})
+
+function getAppByName(apps, appName) {
+  return apps.find(({ name }) => name === appName) || null
+}
+
+function OrgAppsProvider({ children }) {
+  const [apps] = useApps()
+
+  // Avoid additional overhead by finding within existing app list
+  const agreementApp = getAppByName(apps, 'agreement')
+  const disputableVotingApp = getAppByName(apps, 'disputable-voting')
```

Comment:
It felt unnecessary to use `useApp` multiple times when we have to get the full list anyway.